src/core/scope.ml

```diff
@@ -332,7 +345,13 @@ let scope : mode -> sig_state -> env -> p_term -> tbox = fun md ss env t ->
                   r.m_rhs_xvars   <- (id.elt, x) :: r.m_rhs_xvars ;
                   x
         in
-        _TEnv (Bindlib.box_var x) (Array.map (scope env) ts)
+        let ts =
+          match ts with
+          | None -> Env.to_tbox env (* [$M] is equivalent to [$M[env]] where
+                                       [env] is the current environment. *)
+          | Some ts -> Array.map (scope env) ts
```

Comment:
You have twice this piece of code, it can be factorised

Reply:
Not anymore.